crates/vault-registry/src/types.rs

```diff
@@ -699,6 +699,14 @@ impl<T: Config> RichVault<T> {
         }
     }
 
+    pub fn has_no_outstanding_wrapped_tokens(&self) -> Result<bool, DispatchError> {
```

Comment:
Imo having negations in function names can get confusing (especially if you ever need the negation anywhere) - I would flip the condition and go with `has_outstanding_wrapped_tokens`